Unbreak _check_cmdline() on SunOS distributions

eradman opened the pull request:
### What does this PR do?

Prevent PID check from failing on SunOS minions
### What issues does this PR fix or reference?
#34872

(PR #37025 also tries to solve this problem)
### Previous Behavior

```
$ salt database1 cmd.run "sleep 20"
database1:
    Minion did not return. [No response]
```
### New Behavior

```
$ salt database1 cmd.run "sleep 20"
database1:
```
### Tests written?

No, tested manually on SmartOS 5.11 (joyent_20160929T030056Z) on using the
following virtualenv setup

```
pkgin install py27-pip zeromq gcc47 git py27-zmq
pkgin upgrade
virtualenv ~/local/saltenv --system-site-packages
```

Also verified on FreeBSD 11.0-RELEASE-p1

eradman commented:
Accidentally closed this PR by removing the branch in my fork. Reopening

rallytime commented:
Hey @eradman it looks like this will no longer merge cleanly. Can you rebase this with the head of develop?

eradman committed:
Unbreak _check_cmdline() on SunOS distributions

SunOS machines usually have /proc mounted, but it is structured
differently. Run this check only on Linux to ensure reliable operation
on non-Linux hosts.

It is thought that this collision is very rare; log this event on Linux.

Tested on SmartOS 5.11, Fedora 22, MacOS 10.11.4

eradman commented:
Yes, I merged with HEAD and pushed the updated branch

cachedout commented:
There are a number of test failures here on Cent7.

eradman commented:
I cannot easily able to reproduce these failures. I'll see what clues the Jenkins build log shows

    $ python2.7 tests/runtests.py \
    -n integration.renderers.pydsl_test.PyDSLRendererIncludeTestCase.test_rendering_includes \
    -n integration.cli.batch.BatchTest.test_batch_run_grains_targeting \
    -n integration.cli.batch.BatchTest.test_batch_run_number \
    -n integration.netapi.rest_tornado.test_app.TestSaltAPIHandler.test_simple_local_post_only_dictionary_request \
    -n integration.netapi.rest_tornado.test_app.TestSaltAPIHandler.test_simple_local_runner_post \
    -n integration.wheel.key.KeyWheelModuleTest.test_list_all \
    -n integration.cli.grains.GrainsTargetingTest.test_grains_targeting_minion_id_running \
    -n integration.cli.grains.GrainsTargetingTest.test_grains_targeting_os_running \
    -n integration.netapi.test_client.NetapiClientTest.test_local \
    -n integration.shell.matcher.MatchTest.test_ipcidr
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     * Transplanting configuration files to '/tmp/salt-tests-tmpdir/config'
     * Current Directory: /u/en/radman/git/salt-dev
     * Test suite is running under PID 20005
     * Logging tests on /tmp/salt-runtests.log
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     * Setting up Salt daemons to execute tests
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     * Starting salt-master ... STARTED!
     * Starting salt-minion ... STARTED!
     * Starting sub salt-minion ... STARTED!
     * Starting syndic salt-master ... STARTED!
     * Starting salt-syndic ... STARTED!
    =======================================================================================
    
    ...
    
    The Salt Minion is shutdown. Minion received a SIGTERM. Exited.
    The Salt Minion is shutdown. Minion received a SIGTERM. Exited.
    The salt master is shutdown. Master received a SIGTERM. Exited.
    The Salt Syndic is shutdown. Syndic received a SIGINT. Exited.
    
    ==================================  Overall Tests Report  =============================
    ***  No Problems Found While Running Tests  *******************************************
    =======================================================================================
    OK (total=10, skipped=0, passed=10, failures=0, errors=0)
    ==================================  Overall Tests Report  =============================

    $ cat /etc/redhat-release 
    CentOS Linux release 7.1.1503 (Core)

cachedout commented:
Go Go Jenkins!
